test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+RANDOM_SUFFIX=$((RANDOM % 100000 + 1))
+
+PINECONE_ENVIRONMENT="us-east1-gcp"
```

Comment:
Might want `PINECONE_ENVIRONMENT` and `PINECONE_PROJECT_ID` set from CI vars to allow us to point that to different environments/projects with ease.

Reply:
@ryannikolaidis would we want those in CI vars, or would we prefer them explicitly defined in the script? 

I remember us leaving some non-secret CI variables out before, defining them explicitly in scripts. What would we like to do here?

Reply:
We can set a default but allow a custom value to be passed in via CI secrets.

Reply:
This is an interesting question. Looks like CI vars is mostly secrets and tokens. I would lean towards adding less stuff to ci.yml (unless it's sensitive data of course) but we should be consistent so we don't have to think about this. @ahmetmeleq where are the examples of non-secret CI variables defined in script?

Reply:
I agree that fewer variables in CI is better, and we had already decided to leave this kind of variables out of CI in the past. 

I am confused about the change in our approach now, and didn't understand what makes it necessary to put these in CI. I am open to putting these in the CI but I don't want to change approaches often.

Here are the examples I've found to non-secret variables being defined in scripts:

Destination:
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/azure-cognitive-search.sh#L14
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/azure.sh#L19-L23
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/box.sh#L14
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/delta-table.sh#L11
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/dropbox.sh#L13
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/gcs.sh#L13
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/mongodb.sh#L13
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/s3.sh#L13
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/dest/sharepoint-embed-cog-index.sh#L13
Source:
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/confluence-diff.sh#L44
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/jira.sh#L66-L68
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/notion.sh#L39
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/airtable-diff.sh#L30-L31
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/azure.sh#L35-L36
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/delta-table.sh#L42
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/discord.sh#L42
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/dropbox.sh#L49
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/gcs.sh#L49
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/google-drive.sh#L49
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/onedrive.sh#L47
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/outlook.sh#L46
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/s3.sh#L38
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/sharepoint-with-permissions.sh#L55
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/slack.sh#L43
- https://github.com/Unstructured-IO/unstructured/blob/2c2d5b65ca8c2178de33016a4dc331da32bd87c3/test_unstructured_ingest/src/wikipedia.sh#L37

Reply:
I vote: Make the default the current hardcoded variable and optionally set to a value if specified, but let's not bother passing through secrets until we have a compelling reason.

Reply:
Thank you, now addressed with [check if test variables are defined before setting](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/fe818e4767bf4e046bc28efa2ae54d64c62e9c6a)